### DISCUSSION TITLE
Air IOS Publishing Problem with 50.1.1.1

### DISCUSSION THREAD

**Discussion by gmwtrgs** [airsdk/Adobe-Runtime-Support#2353] [General]
Could anyone he,lp with a problem using the latest SDK and Adobe Animate.

I can't create an IPA wi the latest 50.1.1.1 SDK as I just get a whole load of missing symbols errors at the end pf the publishing process in Animate so I tried to roll back to previous release of AIR (50.0.1.3) which gets round this but the IPA it creates is rejected by Apple with a ITMS-90683: Missing purpose string in Info.plist  error.

Does anyone know of a workaround?

**Comment by ajwfrost** [comment]
Sounds like you're using Windows .. we have a problem because of the latest format used by Xcode and iOS16 etc is something that the linker (ld64.exe) on Windows can't cope with. Working on this, we hope to have an update soon. The only workaround currently I'm afraid is to compile on a mac..

thanks

**Reply by marchbold** [reply]
Just a note, I have encountered similar error messages on macOS if you have xcode 13 installed, or the xcode 13 build tools selected using `xcode-select`. Solution: make sure you update to xcode 14.

**Reply by ajwfrost** [reply]
Thanks -> and interestingly, we've just got the latest (available) linker code working on Windows and are still getting these same missing symbols... so we'll need to check further on this, maybe we need to find/include another library or similar..

**Reply by ajwfrost** [reply]
@marchbold  thanks for the hint here .. it looks like it's the compiler that's ended up with differences in the object files, rather than it being a linker problem (well, I guess the linker sorts it out, somehow, so we're not yet sure where the solution will lie...)

From Xcode 13, if we inspect an object file (looking at the `cookieAcceptPolicy` symbol as an example):
```
---------------- d _OBJC_CLASSLIST_REFERENCES_$_
---------------- d _OBJC_CLASSLIST_REFERENCES_$_.12
                 U _OBJC_CLASS_$_NSHTTPCookie
                 U _OBJC_CLASS_$_NSHTTPCookieStorage
---------------- d _OBJC_SELECTOR_REFERENCES_
---------------- d _OBJC_SELECTOR_REFERENCES_.11
---------------- d _OBJC_SELECTOR_REFERENCES_.14
---------------- d _OBJC_SELECTOR_REFERENCES_.16
---------------- d _OBJC_SELECTOR_REFERENCES_.18
---------------- d _OBJC_SELECTOR_REFERENCES_.20
---------------- d _OBJC_SELECTOR_REFERENCES_.9
                 U _objc_msgSend
```
vs in Xcode 14:
```
---------------- d _OBJC_CLASSLIST_REFERENCES_$_
---------------- d _OBJC_CLASSLIST_REFERENCES_$_.8
                 U _OBJC_CLASS_$_NSHTTPCookie
                 U _OBJC_CLASS_$_NSHTTPCookieStorage
                 U _objc_msgSend$cookieAcceptPolicy
                 U _objc_msgSend$cookiesForURL:
                 U _objc_msgSend$keyEnumerator
                 U _objc_msgSend$nextObject
                 U _objc_msgSend$objectForKey:
                 U _objc_msgSend$requestHeaderFieldsWithCookies:
                 U _objc_msgSend$sharedHTTPCookieStorage
```

So, somehow the Apple linker on a mac is managing to work with this new format, but their latest open source version apparently cannot.

There is fortunately a compiler flag that can return the above object symbol references back to how they were before: `-fno-objc-msgsend-selector-stubs`. So if we rebuild with that - and if any ANEs that are built with Xcode 14 also have this flag (in "Other C++ Flags") then we are hopeful it will work...
